packages/shared/src/utils/object-utils.ts

```diff
@@ -77,3 +77,28 @@ export const sortJsonKeys = <T extends Record<string, any>>(json: T): T => {
       return obj;
     }, {}) as T;
 };
+
+/** Minimal deep equality checker, loosely based on lodash _isEqual but for simple primitives only */
+export function isEqual(a: any, b: any) {
+  // handle simple string, boolean, number, undefined, null or same object reference
+  if (a === b) return true;
+  // handle different object types
+  if (typeof a !== typeof b) return false;
+  // handle deep comparison for arrays
+  if (Array.isArray(a)) {
+    // find the first element index where there is a mismatch
+    // if all elements are the same `findIndex` returns value -1
+    const differentIndex = a.findIndex((v, i) => !isEqual(v, b[i]));
+    return differentIndex === -1;
```

Comment:
Nice, this seems like a good way to do it. However, we still need a check that the arrays are of equal length first, otherwise it will return a false positive if `b` contains the same elements as `a` plus some additional elements. E.g. `isEqual([1],[1,2])` will return `true` currently

Reply:
nice point, added length check and updated test with 261a7d96aa565f9b0e36c9db0352a5d1f3ae2d00

Reply:
Also realised that will need to explicitly check `b` is an array as relying on previous `typeof` check could match `object` type for both `[]` or `{}` (or even dates or `null` objects)

Updated with b43d534a7e1efc25c92c5508deb7645eea4ce1d4